storage/url/url.go

```diff
@@ -178,7 +179,11 @@ func (u *URL) remoteURL() string {
 	}
 
 	if u.Path != "" {
-		s += "/" + u.Path
+		pathElements := regexp.MustCompile(`/`).Split(u.Path, -1)
```

Comment:
S3 is returning `404 Not Found` because s5cmd is not escaping the s3 key. 

Could you please add escape logic to `s3.go` where we are creating `s3.CopyObjectInput` ?
https://github.com/peak/s5cmd/blob/6b97a9eeb01c4d7a2658613aead539702197bbd4/storage/s3.go#L319

Updaing `url.go` may break wildcard support.

Reply:
Hi @ilkinulas
I updated as your suggestion.

---

## About the test

I tested with the real s3 buckets, and it works.

I tried a new unit test. However, it doesn't work as expected. 

- If I `putFile()` an object with key: `a+1.txt`, running the cmd `cp s3://srcBucket/* s3://dstBucket/` will throw an 404 error because the key `a%2B1.txt` doesn't exist.
- The same error with key `a%2B1.txt` is sent to `putFile()`. Because the key is not converted to `a+1.txt` as the real S3 behavior. So, executing the `cp` command, the object key will be escaped one more time to `a%252B1.txt`

I think that's why the test `TestRunSpecialCharactersInPrefix()` was passed.

Reply:
I have also tried to write a test that uses `putFile(...)` and test passed. Could you please share your test where copy command throws an 404 error?

Reply:
@ilkinulas 

I pushed my test. Here are the error:

---
## Case 1:

<img width="1312" alt="Screen Shot 2021-04-08 at 23 38 17" src="https://user-images.githubusercontent.com/13713865/114065853-4cd5ee80-98c5-11eb-932d-5cb5ec9f09ae.png">

the following errors occurs:

```txt
Stderr:   ERROR "cp s3://bucket/sub&@$/test+1.txt s3://dstbucket/sub&@$/test+1.txt": NoSuchKey:  status code: 404, request id: 000000000000000A, host id: MDAwMDAwMDAwMDAwMDAwQTAwMDAwMDAwMDAwMDAwMEEwMDAwMDAwMDAwMDAwMDBBMDAwMDAwMDAwMDAwMDAwQQ==
        ERROR "cp s3://bucket/test&@$:,?;= 3.txt s3://dstbucket/test&@$:,?;= 3.txt": NoSuchKey:  status code: 404, request id: 000000000000000B, host id: MDAwMDAwMDAwMDAwMDAwQjAwMDAwMDAwMDAwMDAwMEIwMDAwMDAwMDAwMDAwMDBCMDAwMDAwMDAwMDAwMDAwQg==
        ERROR "cp s3://bucket/sub:,?/test; =2.txt s3://dstbucket/sub:,?/test; =2.txt": NoSuchKey:  status code: 404, request id: 000000000000000D, host id: MDAwMDAwMDAwMDAwMDAwRDAwMDAwMDAwMDAwMDAwMEQwMDAwMDAwMDAwMDAwMDBEMDAwMDAwMDAwMDAwMDAwRA==
```

---

## Case 2:

<img width="1296" alt="Screen Shot 2021-04-08 at 23 40 05" src="https://user-images.githubusercontent.com/13713865/114066170-9e7e7900-98c5-11eb-91ac-24b732397c34.png">

the following errors occurs:

```txt
Stderr:   ERROR "cp s3://bucket/sub%3A%2C%3F/test%3B+%3D2.txt s3://dstbucket/sub%3A%2C%3F/test%3B+%3D2.txt": NoSuchKey:  status code: 404, request id: 000000000000000A, host id: MDAwMDAwMDAwMDAwMDAwQTAwMDAwMDAwMDAwMDAwMEEwMDAwMDAwMDAwMDAwMDBBMDAwMDAwMDAwMDAwMDAwQQ==
        ERROR "cp s3://bucket/test%26%40%24%3A%2C%3F%3B%3D+3.txt s3://dstbucket/test%26%40%24%3A%2C%3F%3B%3D+3.txt": NoSuchKey:  status code: 404, request id: 000000000000000B, host id: MDAwMDAwMDAwMDAwMDAwQjAwMDAwMDAwMDAwMDAwMEIwMDAwMDAwMDAwMDAwMDBCMDAwMDAwMDAwMDAwMDAwQg==
        ERROR "cp s3://bucket/sub%26%40%24/test%2B1.txt s3://dstbucket/sub%26%40%24/test%2B1.txt": NoSuchKey:  status code: 404, request id: 000000000000000C, host id: MDAwMDAwMDAwMDAwMDAwQzAwMDAwMDAwMDAwMDAwMEMwMDAwMDAwMDAwMDAwMDBDMDAwMDAwMDAwMDAwMDAwQw==
```

Reply:
I am still trying to find out why test is failing with 404, fyi